pkl-core/src/main/java/org/pkl/core/module/ResolvedModuleKeys.java

```diff
@@ -75,7 +77,16 @@ public URI getUri() {
 
     @Override
     public String loadSource() throws IOException {
-      return Files.readString(path, StandardCharsets.UTF_8);
+      try {
+        return Files.readString(path, StandardCharsets.UTF_8);
+      } catch (AccessDeniedException e) {
+        // Windows throws `AccessDeniedException` when reading directories.
+        // Sync error between different OSes.
+        if (Files.isDirectory(path)) {
+          throw new IOException("Is a directory");
```

Comment:
Should we not preserve the trace? I'd, at least, add `e` as a cause.
```suggestion
          throw new IOException("Is a directory", e);
```

Reply:
While this is generally a good idea, it works against us here because the cause gets rendered in the `-- Pkl Error --` message. The idea here is that we want to make this error message identical on the two platforms.